micro-rdk/src/esp32/conn/network.rs

```diff
@@ -0,0 +1,134 @@
+use std::{
+    net::Ipv4Addr,
+    sync::{Arc, Mutex},
+};
+
+#[cfg(not(feature = "qemu"))]
+use {
+    crate::common::conn::network::{Network, NetworkError},
+    crate::esp32::esp_idf_svc::{
+        eventloop::{EspSubscription, EspSystemEventLoop, System},
+        hal::modem::Modem,
+        sys::esp_wifi_set_ps,
+        wifi::{BlockingWifi, EspWifi, WifiEvent},
+    },
+    embedded_svc::wifi::{
+        AuthMethod, ClientConfiguration as WifiClientConfiguration,
+        Configuration as WifiConfiguration,
+    },
+};
+
+use crate::esp32::esp_idf_svc::{
+    eth::{BlockingEth, EspEth, OpenEth},
+    sys::EspError,
+};
+
+/// A wrapper around the wifi structure available in esp-idf-svc with and adjustment to support
+/// reconnection
+#[cfg(not(feature = "qemu"))]
+pub struct Esp32WifiNetwork {
+    inner: Arc<Mutex<Box<BlockingWifi<EspWifi<'static>>>>>,
+    sl_stack: EspSystemEventLoop,
+    _subscription: Option<EspSubscription<'static, System>>,
+}
+
+#[cfg(not(feature = "qemu"))]
```

Comment:
don't forget to remove qemu feature inside micro-rdk

Reply:
oh I misinterpreted what you had asked. Yeah will do